### ISSUE TITLE
Unable to debug MAUI application using VS Code for Mac on iPhone

### ISSUE THREAD

**Issue by iSukhi** [xamarin/xamarin-macios#21664]
I am using VS Code for Mac with the following specification.

Version: 1.95.3 (Universal)
Commit: f1a4fb101478ce6ec82fe9627c43efbf9e98c813
Date: 2024-11-13T14:50:04.152Z (6 days ago)
Electron: 32.2.1
ElectronBuildId: 10427718
Chromium: 128.0.6613.186
Node.js: 20.18.0
V8: 12.8.374.38-electron.0
OS: Darwin arm64 24.1.0

The application gets installed on iPhone, but I am not able to debug the application on iPhone (No breakpoints are hitting)
When I am deploying the application I am getting the following log:

```
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net9.0_18.0/18.0.9617/tools/bin/mlaunch
error MT0000: Unexpected error - Please file a bug report at https://github.com/xamarin/xamarin-macios/issues/new
System.NotSupportedException: Specified method is not supported.
at Xamarin.Hosting.DeviceCtlDevice.CreateTcpTunnel(LaunchConfig) in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Hosting/DeviceCtlDevice.cs:line 79
at Xamarin.Launcher.DevController.CreateTcpTunnel(String, LaunchConfig) in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/controller-device.cs:line 844
at Xamarin.Launcher.Driver.MainAsync() in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/Main.cs:line 536
at Xamarin.Utils.NSRunLoopExtensions.RunUntilTaskCompletion[T](NSRunLoop, Task`1) in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Utils/Extensions.cs:line 30
at Xamarin.Launcher.Driver.Main2(String[]) in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/Main.cs:line 443
at Xamarin.Launcher.Driver.Main(String[]) in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/Main.cs:line 130
error MT0000: Unexpected error - Please file a bug report at https://github.com/xamarin/xamarin-macios/issues/new
System.NotSupportedException: Specified method is not supported.
at Xamarin.Hosting.DeviceCtlDevice.CreateTcpTunnel(LaunchConfig) in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Hosting/DeviceCtlDevice.cs:line 79
at Xamarin.Launcher.DevController.CreateTcpTunnel(String, LaunchConfig) in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/controller-device.cs:line 844
at Xamarin.Launcher.Driver.MainAsync() in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/Main.cs:line 536
at Xamarin.Utils.NSRunLoopExtensions.RunUntilTaskCompletion[T](NSRunLoop, Task`1) in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Utils/Extensions.cs:line 30
at Xamarin.Launcher.Driver.Main2(String[]) in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/Main.cs:line 443
at Xamarin.Launcher.Driver.Main(String[]) in /Users/builder/azdo/_work/2/s/tools/mlaunch/Xamarin.Hosting/Xamarin.Launcher/Main.cs:line 130
```

**Comment by rolfbjarne** [comment]
Does it work if you use VS Code instead of VS Mac?

**Comment by iSukhi** [comment]
@rolfbjarne
I am using VS Code only I am not using VS for MAC as it is retired.

**Comment by iSukhi** [comment]
@rolfbjarne I am using VS Code only

**Comment by iSukhi** [comment]
> Does it work if you use VS Code instead of VS Mac?

**Comment by vaibhav-sukhdeve** [comment]
@rolfbjarne If I may ask when this will take effect in VS code?
Will it come as an update for IDE or MAUI workload?

**Comment by mlancione** [comment]
I am running into this issue. It looks like this change was not included in today's 18.1.9163 service release:
https://github.com/xamarin/xamarin-macios/pull/21673

Any way for us to deploy the change ourselves manually?

**Comment by JaneySprings** [comment]
Hi, @rolfbjarne! I'm a creator of [DotNet.Meteor](https://marketplace.visualstudio.com/items?itemName=nromanov.dotnet-meteor) extension for VSCode, I have the same issue with `mlaunch --tcp-tunnel`.

How to use `mlaunch` now for creating a tcp tunnel to device? This is my old way:

https://github.com/JaneySprings/DotNet.Meteor/blob/de3f722be7781bbae8dc38f6d4b8ed152d077275/src/DotNet.Meteor.Common/Apple/MonoLauncher.cs#L7-L13
```
mlaunch --tcp-tunnel={port}:{port} --devname={serial}
```

**Comment by rolfbjarne** [comment]
> I am running into this issue. It looks like this change was not included in today's 18.1.9163 service release: #21673
> 
> Any way for us to deploy the change ourselves manually?

Yes, I believe so, although it's a bit cumbersome:

1. Go here and click "Download" to download the Microsoft.Tools.Mlaunch NuGet: https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet-eng/NuGet/Microsoft.Tools.Mlaunch/overview/1.0.273%201160e18b07

2. Rename the file from `Microsoft.Tools.Mlaunch.1.0.273.nupkg` to `Microsoft.Tools.Mlaunch.1.0.273.zip`
3. Extract everything in the zip file somewhere.
4. Move this directory from the zip file:

    > mlaunch/lib/mlaunch/mlaunch.app

    into this directory (note that the exact path depends on which version of the iOS workload you have installed):
  
    > /usr/local/share/dotnet/packs/Microsoft.iOS.Sdk.net9.0_18.0/18.0.9617/tools/lib/mlaunch/

   (overwriting the existing mlaunch.app directory there)

**Comment by rolfbjarne** [comment]
> Hi, @rolfbjarne! I'm a creator of [DotNet.Meteor](https://marketplace.visualstudio.com/items?itemName=nromanov.dotnet-meteor) extension for VSCode, I have the same issue with `mlaunch --tcp-tunnel`.
> 
> How to use `mlaunch` now for creating a tcp tunnel to device? This is my old way:
> 
> [JaneySprings/DotNet.Meteor@`de3f722`/src/DotNet.Meteor.Common/Apple/MonoLauncher.cs#L7-L13](https://github.com/JaneySprings/DotNet.Meteor/blob/de3f722be7781bbae8dc38f6d4b8ed152d077275/src/DotNet.Meteor.Common/Apple/MonoLauncher.cs#L7-L13)
> 
> ```
> mlaunch --tcp-tunnel={port}:{port} --devname={serial}
> ```

That should work once the fix I made has been released.
